### DISCUSSION TITLE
Source Control Widget Not Working Properly

### DISCUSSION THREAD

**Discussion by cj-rajat** [eclipse-theia/theia#11701] [Q&A]
Hi I tried the same [way](https://github.com/eclipse-theia/theia/discussions/10753) I removed the @theia/git and @theia/scm-extra and install vscode builtin git extensions(vscode.git, vscode.git-ui), but here I am getting the errors if I take any action from source control widget like clicking on tick icon or more actions menu.

![image](https://user-images.githubusercontent.com/84508727/192315279-a3d11414-d348-4ac7-9acc-0e9b6b0dde36.png)

also there are some warnings in the console

![image](https://user-images.githubusercontent.com/84508727/192315778-10cdd023-ed1e-4bc5-b35c-4380611966e4.png)

Also If I clone the repo using Command Palette -> Git Clone, then it asked for GitHub authentication, and when I click on allow there is error Error: authProvider.createSession is not a function

**Comment by tyagirajat200** [comment]
Any Update ?

**Comment by vince-fugnitto** [comment]
@rajattyagi200 thank you for the discussion, I've opened https://github.com/eclipse-theia/theia/issues/11730 to track implementing the missing menu paths.

**Reply by tyagirajat200** [reply]
Thanks, 
Is there a way to remove this source control view completely from the left container UI ?

Also from the explorer view container, how we can remove the NPM Scripts widget ?

**Reply by vince-fugnitto** [reply]
@rajattyagi200 

> Is there a way to remove this source control view completely from the left container UI ?

Given that `@theia/scm` is pulled as a dependency from the plugin system, you'll need to filter out it's frontend contribution like the following: https://community.theia-ide.org/t/i-need-to-remove-debug-functionality/2602/2.

> Also from the explorer view container, how we can remove the NPM Scripts widget ?

The `npm scripts` tree-view item comes from a plugin (https://open-vsx.org/extension/vscode/npm), you can either:
- remove the plugin if  you do not need it
- maintain your own version of the plugin without the tree-view
- customize our tree-view handling to exclude the view

Note that you can hide the view for yourself, but it seems like you want to remove it completely.
